LibraBFT/ImplShared/Util/RWST/Syntax.agda

```diff
@@ -60,6 +60,12 @@ ifM b then c₁ else c₂ =
   ifM‖ b ≔ c₁
      ‖ otherwise≔ c₂
 
+ifMHask : ⦃ _ : ToBool B ⦄ → RWST Ev Wr St B → (c₁ c₂ : RWST Ev Wr St A) → RWST Ev Wr St A
```

Comment:
@mark-moir I anticipate that for proofs we will want `ifMHask` to use the `RWST` constructor. This poses a bit of a problem.
- If in the Haskell code we use `ifM` in e.g. the `Either (ErrLog a)` monad, we will want a version of `ifMHask` (eventually to be renamed to `ifM`) that works for any monad. This suggests the need for an `ifMHask` variant for `RWST`
- If not, we can get by with an `ifMHask` that only works for `RWST` (so, taking this implementation and updating it to use the `RWST` constructor for `if`)

Reply:
This makes sense @cwjnkins, thanks.  If we do find that we need a more general `ifM`, we can add that to Prelude and then perhaps keep this one as `ifM` but use it qualified, as in `RWST-Syntax.ifM`, for example.  Anyway, I'm happy with the current approach (as already implemented by @haroldcarr) until then.  Thanks.